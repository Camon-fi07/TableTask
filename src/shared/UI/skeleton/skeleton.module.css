.skeleton {
  background: var(--colorSkeleton);
  background-size: 200%;
  animation: shimmer 1.5s infinite linear;
  width: 100px;
  height: 25px;
  border-radius: 4px;
}

@keyframes shimmer {
  0% {
    background-position: 200%;
  }
  100% {
    background-position: -200%;
  }
}
