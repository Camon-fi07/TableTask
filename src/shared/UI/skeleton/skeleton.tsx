import clsx from 'clsx';
import { CSSProperties } from 'react';

import styles from './skeleton.module.css';

interface SkeletonProps {
  width?: CSSProperties['width'];
  height: CSSProperties['height'];
  margin?: CSSProperties['margin'];
  borderRadius: CSSProperties['borderRadius'];
  className?: string;
}

export const Skeleton = ({
  borderRadius,
  height,
  className,
  margin,
  width
}: SkeletonProps) => (
  <div
    className={clsx(styles.skeleton, className)}
    style={{ height, width, margin, borderRadius }}
  />
);
