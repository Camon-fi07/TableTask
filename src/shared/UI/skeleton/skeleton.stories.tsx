import type { Meta, StoryObj } from '@storybook/react-vite';

import { Skeleton } from './skeleton';

const meta: Meta<typeof Skeleton> = {
  title: 'UI/Skeleton',
  component: Skeleton,
  tags: ['autodocs'],
  argTypes: {}
};

export default meta;
type Story = StoryObj<typeof Skeleton>;

export const Default: Story = {
  args: {}
};

export const EditSize: Story = {
  args: {
    borderRadius: '50%',
    width: 30,
    height: 30
  }
};
