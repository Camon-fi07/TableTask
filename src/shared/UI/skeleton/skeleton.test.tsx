import { render, screen } from '@testing-library/react';

import { Skeleton } from './skeleton';

describe('Skeleton', () => {
  it('should render', () => {
    render(<Skeleton />);
    expect(screen.getByTestId('skeleton')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    render(<Skeleton />);
    expect(screen.getByTestId('skeleton')).toMatchSnapshot();
  });

  it('should apply css properties', () => {
    render(<Skeleton borderRadius='50%' width={30} height={30} margin={10} />);

    expect(screen.getByTestId('skeleton')).toHaveStyle('border-radius: 50%');
    expect(screen.getByTestId('skeleton')).toHaveStyle('width: 30px');
    expect(screen.getByTestId('skeleton')).toHaveStyle('height: 30px');
    expect(screen.getByTestId('skeleton')).toHaveStyle('margin: 10px');
    expect(screen.getByTestId('skeleton')).toMatchSnapshot();
  });
});
