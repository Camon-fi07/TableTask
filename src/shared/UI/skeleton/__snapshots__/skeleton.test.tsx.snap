// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should apply css properties 1`] = `
<div
  class="skeleton"
  data-testid="skeleton"
  style="height: 30px; width: 30px; margin: 10px; border-radius: 50%;"
/>
`;

exports[`Skeleton should match snapshot 1`] = `
<div
  class="skeleton"
  data-testid="skeleton"
/>
`;
